### PR TITLE
added estimateTranslation3D to calib3d/ptsetreg

### PR DESCRIPTION
**Pull request by GFleishman**
I've done my best to follow the PR guidelines, but this is my first PR to OpenCV, so I anticipate some back and forth; I'm grateful for your time.

Currently the RANSAC filter in calib3D/ptsetreg.cpp only permits full affine transformations. I have an application (and others may as well) that requires limiting the transform to translation only.

The new function follows the API and implementation structure for estimateAffine3D exactly, only the implementations of runKernel, computeError, and checkSubset are modified (and are considerably simpler) for the new class - users can call estimateTranslation3D as a drop in replacement for estimateAffine3D whenever it is appropriate.

I've built successfully on my system and implemented tests. estimateTranslation3D will be strictly faster than estimateAffine3D as it does not require SVD.

I believe I have documented the new code properly, as I simply copied the documentation strings from estimateAffine3D and modified them to be consistent with translation only. Please advise if I'm missing anything.

### Pull Request Readiness Checklist

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [X] I agree to contribute to the project under OpenCV (BSD) License.
- [X] To the best of my knowledge, the proposed patch is not based on a code under GPL or other license that is incompatible with OpenCV
- [X] The PR is proposed to proper branch
- [NA] There is reference to original bug report and related work
- [X] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [X] The feature is well documented and sample code can be built with the project CMake

### CODE REVIEW

**Comment by GFleishman** [comment]
I believe the build error is because of warnings only. The warnings are because checkSubset in my implementation of Translation3DEstimatorCallback does not use the input arguments (i.e. unused arguments errors). But checkSubset must be implemented with these arguments to be consistent with objects which use Translation3DEstimatorCallback (namely RANSACPointSetRegistrator). Not sure of the best way to resolve this.

**Comment by asmorkalov** [comment]
@GFleishman Good job! Could you add tests for the solution you introduced?

**Comment by GFleishman** [comment]
@asmorkalov Yep! I saw the tags last week but have been very busy. I think just copying and renaming the existing tests for estimateAffine3D will probably work. I may get to this today.

**Comment by GFleishman** [comment]
@asmorkalov Ok! I've added `test_translation3d_estimator.cpp` to `modules/calib3d/test`
It compiles just fine, and when I execute the binary `opencv_test_calib3d` the new tests are included in the run and pass with OK. It was a very straight forward mapping of `test_affine3d_estimator.cpp`
Please let me know if there is anything else I should do to get this PR merged :)

**Comment by asmorkalov** [comment]
@GFleishman Good job! Thanks for tests.

**Comment by asmorkalov** [comment]
@vpisarev could you look on the PR?

**Comment by vpisarev** [comment]
:+1:

**Comment by vpisarev** [comment]
Great job, thank you!

File: modules/calib3d/src/ptsetreg.cpp [hunk 1]
```diff
@@ -505,6 +505,93 @@ class Affine3DEstimatorCallback : public PointSetRegistrator::Callback
...
+    int runKernel( InputArray _m1, InputArray _m2, OutputArray _model ) const CV_OVERRIDE
+    {
+
+        Mat m1 = _m1.getMat(), m2 = _m2.getMat();
+        const Point3f* from = m1.ptr<Point3f>();
+        const Point3f* to   = m2.ptr<Point3f>();
+
+        double buf[3];
+        Mat T(3, 1, CV_64F, &buf[0]);
+        double* Tptr = T.ptr<double>();
```
Comment (alalek) [inline modules/calib3d/src/ptsetreg.cpp]: Consider using `Matx13d` instead.

```.diff
-T.copyTo(_model);
+Mat(T, false).copyTo(_model);
```

File: modules/calib3d/src/ptsetreg.cpp [hunk 2]
```diff
@@ -505,6 +505,93 @@ class Affine3DEstimatorCallback : public PointSetRegistrator::Callback
...
+            const Point3f& f = from[i];
+            const Point3f& t = to[i];
+
+            Tptr[0] = Tptr[0] + t.x - f.x;
+            Tptr[1] = Tptr[1] + t.y - f.y;
+            Tptr[2] = Tptr[2] + t.z - f.z;
+        }
+        Tptr[0] = Tptr[0] / 4;
+        Tptr[1] = Tptr[1] / 4;
+        Tptr[2] = Tptr[2] / 4;
```
Comment (alalek) [inline modules/calib3d/src/ptsetreg.cpp]: `T *= (1.0f / 4);` ?

File: modules/calib3d/test/test_translation3d_estimator.cpp [hunk 3]
```diff
@@ -0,0 +1,211 @@
+/*M///////////////////////////////////////////////////////////////////////////////////////
+//
+//  IMPORTANT: READ BEFORE DOWNLOADING, COPYING, INSTALLING OR USING.
```
[the remark this answers is not in the document]
Comment (GFleishman) [reply modules/calib3d/test/test_translation3d_estimator.cpp]: Done!

File: modules/calib3d/src/ptsetreg.cpp (hunk 1 above)
Comment (GFleishman) [reply modules/calib3d/src/ptsetreg.cpp]: Done! Also required modifying `test_translation3d_estimator` to use the new data type

File: modules/calib3d/src/ptsetreg.cpp (hunk 2 above)
Comment (GFleishman) [reply modules/calib3d/src/ptsetreg.cpp]: Done! More concise 👍

**Comment by GFleishman** [comment]
Hi guys - what is still required for merging?

**Comment by alalek** [comment]
OpenCV code base is in code freeze state due release preparation. This patch will be merged after 3.4.10 release (targeted to 3.4.11).

**Pull request merged**